### DISCUSSION TITLE
Need some suggestions in migrating existing framework from selenium to selenide

### DISCUSSION THREAD

**Discussion by aravindraveendran7** [selenide/selenide#2376] [General]
Hello,
In my organization, we currently use Selenium to automate UI applications. I would like to check with the community that, Is it a good practice to use selenide methods like the different wait conditions in places where required and avoiding custom wait strategies from selenium to reduce boilerplate codes. Basically, can I use both selenium and selenide hand in hand, as currently the driver initialization and browser launching are done via selenium explicitly?

**Comment by asolntsev** [comment]
Hi @aravindraveendran7 !
1. Yes, it's it's a good practice to use Selenide methods (`$.shouldHave`, `$.shouldBe` etc.). Not only to reduce boilerplate code, but also to make the test more stable (and sometimes faster).
2. Yes, you can use both Selenide and Selenium hand-in-hand. You can leave your browser initialisation code as is, and just ask Selenide to use the same webdriver (using method `WebDriverRunner.setWebDriver(yourDriver)`). 
Then you can write only new pieces of code on Selenide. And maybe rewrite some old pieces to Selenide if needed.
